### ISSUE TITLE
Make links behave consistently on node page with comments

### ISSUE THREAD

**Issue by stpaultim** [backdrop/backdrop-issues#4670]
**Description of the bug**

Followup to #https://github.com/backdrop/backdrop-issues/issues/4203

In that issue, we added (are adding) an underline to some specific links on the comments box. In reviewing that issue, I found that other links on the same page behave very inconsistently. 

I'm not sure how all of them should behave. Part of this issue might be to write up a little style guide about how links should work. 

At a minimum, let's get all the links on this page working properly. The user "admin" has two links on this page, one starts with underline that disappears on hover, the other starts without underline and it appears on hover. That can't be right. 

**Steps To Reproduce**

1) Spin up a demo site at https://backdropcms.org/
2) Login and create a new post: node/add/post
3) Save post and then view post. 
4) Add a comment and save comment. 
5) View post again and notice inconsistent behavior of links on the page.  

**Actual behavior**

Inconsistent behavior of links on the page. 

**Expected behavior**

That the links on the page behave consistently as one might expect them too. I'm not sure if from a usability some of these links are different than the others and expected to behave differently, but we should decide how they are expected to behave and make that behave that way. 

Do we have styeguilde for Backdrop CMS core that might provide some guidance on this issue?

**Additional information**

![Your_first_post____PR_3333_backdrop_backdrop_testing_site](https://user-images.githubusercontent.com/3144571/94773889-54e72b80-0382-11eb-8af3-3ff9da0ea120.png)

On the node edit page, links seem to act more as I would expect them to. 

** Possible Followup issue **

Move around the Backdrop CMS interface and look for other examples of misbehaving links - either open new issues or fix them with this one.

**Comment by BWPanda** [comment]
Wow, this is really messed up! :smile:

**Comment by jenlampton** [comment]
> Do we have styeguilde for Backdrop CMS core that might provide some guidance on this issue?

I think we had one at one point, but I don't know where this resource lives. @wesruv do you remember what we used as a styleguide for Basis? Was it a PDF? A living html/css document? 

The code below is from `skin.css` in Basis.

```css
a:focus,
a:hover {
  text-decoration: none;
}
```

This would lead me to believe that links are *not* supposed to be underlined on hover. I expect that there is other CSS being added from various parts of core that is overriding the intent of basis, in different ways. 

I personally feel that websites are more usable when there is some indication that the text you are hovering over is a link. If it were up to me. I would modify Basis so that all links consistently had underlines when on `:hover` and `:active`.

Would this be too much of a change?

**Comment by stpaultim** [comment]
Again, anyone should feel free to follow the steps to reproduce (in original post) and add their opinion to this discussion. How do you expect these various links to behave?

Also - is there any risk to existing sites if we try to change these behaviors?

**Comment by wesruv** [comment]
I'm a big fan of obvious hover and obvious focus is a requirement for accessibility. I'd guess that either:
* The underline is supposed to be there by default and removed for hover?
* I was using border or box shadow instead if text decoration?

But that's a wild guess, could be past me goofed or something changed between now and then.

**Comment by jenlampton** [comment]
> The underline is supposed to be there by default and removed for hover?

I think this is a straightforward solution. 

I see the following in `base.css`:

```css
h1 a,
h2 a,
h3 a,
h4 a,
h5 a,
h6 a {
  text-decoration: none;
}
```

I think we should make the inverse true for hearings, no underline by default, but add the underline for hover/focus. So we'd need to add...

```css
h1 a:hover,
h2 a:hover,
h3 a:hover,
h4 a:hover,
h5 a:hover,
h6 a:hover {
  text-decoration: underline;
}
```

**Comment by stpaultim** [comment]
@jenlampton and I just chatted about this and are recommending that someone create a Pull Request with the above changes. We really want to see how this will effect other things and we may need make additional modifications later. 

Still a great issue for a OSD participant.

**Comment by BWPanda** [comment]
I personally feel that links should stand out from regular text somehow, and that they should visually change when hovered/focused on.

Generally this means either:

1. Links are not underlined but are a different colour to regular text (stand out) and then the underline is added on hover/focus (visual change).
2. Links are the same colour as regular text but are underlined (stand out) and then the underline is removed on hover/focus (visual change).

Either of those solutions are fine by me, as long as they're used consistently (what we're currently lacking).

**Comment by oadaeh** [comment]
> 2. Links are the same colour as regular text but are underlined (stand out) and then the underline is removed on hover/focus (visual change).

I'm not so sure that is a good idea. I have had times when I accidentally/inadvertently left/moved the pointer over a link without realizing it, and then when I try visually looking for it, I don't notice it until I move the mouse again. I wonder also if that would create an accessibility problem.

**Comment by Sahana-Pandurangi** [comment]
I would like to work on this issue

**Comment by jenlampton** [comment]
I still think we should go with @wesruv's recommendation from above (A below), which would be a combination of both options 1 & 2 mentioned in @BWPanda's list.

A. @wesruv Links are a different colour from normal text, and ARE underlined. On hover, the underline is removed. 
B. @bwpanda Links a different colour from regular text, and are NOT underlined. On hover, the underline is added.
C. @bwpanda Links are the same colour as regular text but ARE underlined. On hover, the underline is removed.

**Comment by tanisha-tech360** [comment]
Hey  I am an OpenForce 2022 Contributor . I would like to work on this issue. I would be making a PR as soon as I am done with resolving the issue. Thank you"

**Comment by stpaultim** [comment]
@tanisha-tech360 - Welcome to the Backdrop CMS issue queue. Please, feel free to post questions here. We would like to help you solve this problem.

**Comment by stpaultim** [comment]
@tanisha-tech360 - Are you still interested in working on this issue? Do you have any questions?

**Comment by tanisha-tech360** [comment]
Yes, I am still working on this issue. Actually I am a beginner can you tell give me some resources.

**Comment by tanisha-tech360** [comment]
Please check pull request [#4000](https://github.com/backdrop/backdrop/pull/4000)

**Comment by stpaultim** [comment]
@tanisha-tech360 - I don't think your PR is doing the job yet, I believe it will need a little more work. 

You can test your PR, by going to the PR page and logging into the sandbox. 
![image](https://user-images.githubusercontent.com/3144571/158958955-e06d8ebb-1330-4988-93d2-b3a77c064cb5.png)

Once in the Sandbox, follow the steps to recreate the issue listed above (in the initial post). I think you will notice that many of the same inconsistencies are still present. The goal here is to make sure that links are behaving in a consistent manner.

I think this is a good place to start: Links are a different colour from normal text, and ARE underlined. On hover, the underline is removed.

**Comment by stpaultim** [comment]
I changed the title, because @jenlampton suggested in her last comment that we go with:

A. @wesruv Links are a different colour from normal text, and ARE underlined. On hover, the underline is removed.

But, the title said the opposite.
